src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -293,6 +370,14 @@ protected Query doToQuery(QueryShardContext context) {
         } else if (knnMethodContext != null) {
```

Comment:
spaceType should be retrievable from modelMetadata as well. If we default to SpaceType.DEFAULT, this could lead to silent functional failure

Reply:
Added retrievable from modelMetadata, and we probably need keep the initial default spaceType similar to knnEngine, which is mainly used in tests.